docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+    clusterAutoscaler:
+      expander: "least-waste"
+      scanInterval: 5s
+      scaleDownDelayAfterAdd: 1m
+      scaleDownDelayAfterDelete: 0s
+      scaleDownDelayAfterFailure: 1m
+      scaleDownUnneededTime: 1m
+      scaleDownUtilizationThreshold: 0.5
+  provider:
+    workers:
+    - name: ...
+      minimum: 6
+      maximum: 60
+      maxSurge: 3
+      maxUnavailable: 0
+      zones:
+      - ... # list of zones you want your worker pool nodes to be spread across, see above
+      kubernetes:
+        kubelet:
+          ... # similar to `kubelet` above (cluster-wide settings), but here per worker pool (pool-specific settings), see above
+      machineControllerManager: # optional, it allows to configure the machine-controller settings.
+        machineCreationTimeout: 10m
+        machineHealthTimeout: 2m
+        machineDrainTimeout: 5m
+  systemComponents:
+    coreDNS:
+      autoscaling:
+        mode: horizontal # valid values are `horizontal` (driven by CPU load) and `cluster-proportional` (driven by number of nodes/cores)
+```
+
+#### On `spec.controlPlane.highAvailability.failureTolerance.type`
+
+If set, determines the degree of failure tolerance for your control plane. `zone` is preferred, but only available if your control plane resides in a region with 3+ zones. See [above](#control-plane) and the [docs](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_high_availability.md).
+
+#### On `spec.kubernetes.kubeAPIServer.defaultUnreachableTolerationSeconds` and `defaultNotReadyTolerationSeconds`
+
+This is a very interesting [API server setting](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver) that lets Kubernetes decide how fast to evict pods from nodes whose status condition of type `Ready` is either `Unknown` (node status unknown, a.k.a unreachable) or `False` (`kubelet` not ready) (see [node status conditions](https://kubernetes.io/docs/concepts/architecture/nodes/#condition); please note that `kubectl` shows both values as `NotReady` which is a somewhat "simplified" visualization).
+
+You can also override the cluster-wide API server settings [individually per pod](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions):
+
+``` yaml
+spec:
+  tolerations:
+  - key: "node.kubernetes.io/unreachable"
+    operator: "Exists"
+    effect: "NoExecute"
+    tolerationSeconds: 0
+  - key: "node.kubernetes.io/not-ready"
+    operator: "Exists"
+    effect: "NoExecute"
+    tolerationSeconds: 0
+```
+
+This will evict pods on unreachable or not-ready nodes immediately, but be cautious: `0` is very aggressive and may lead to unnecessary disruptions. Again, you must decide for your own workload and balance out the pros and cons (e.g. long startup time).
+
+Please note, these settings replace the deprecated `spec.kubernetes.kubeControllerManager.podEvictionTimeout` that was removed with Kubernetes `v1.26` (and acted as an upper bound).
```

Comment:
Nit: The `--pod-eviction-timeout` flag was deprecated in K8s 1.26 with https://github.com/kubernetes/kubernetes/pull/113710. There are several open PRs to remove it for 1.27.

```suggestion
Please note, these settings replace the `spec.kubernetes.kubeControllerManager.podEvictionTimeout` field that was deprecated in Kubernetes `v1.26` (and acted as an upper bound).
```

Reply:
Thanks. I only checked the KCM options and they were gone in the docs, but that's not what I wrote. I will improve as suggested.